### ISSUE TITLE
use safe math in allowance operator

### ISSUE THREAD

**Issue by manuelpolzhofer** [centrifuge/tinlake#302]
use `safeMath` functions in allowance operator

https://github.com/centrifuge/tinlake/blob/develop/src/lender/tranche/operator/allowance.sol

**Comment by philipstanislaus** [comment]
@manuelpolzhofer @ilinzweilin why don't we just use `using SafeMath for uint;` for all contracts as the default? (I have very little experience with writing Smart Contracts – just wondering ;) ) https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/math/SafeMath.sol

**Comment by manuelpolzhofer** [comment]
We made your own safeMath lib and re-used the implementations from Maker because it is a bit more gas efficient because we don't need the `mod` operations from open zeppelin. 

We tried to avoid libraries in the codebase for security reasons.  The `using for` operator is based on libraries.
https://solidity.readthedocs.io/en/develop/contracts.html#using-for

But, you are right maybe we should take a look again. For safeMath it might make sense 👍 
I need to understand the `using for` a bit better
